### PR TITLE
Korean changes

### LINKED ISSUE #4: Korean Layout Doesn't work
Hello, a friend of mine and I happened to fix the issue with the Korean layout not properly creating Korean characters (Hangul). We were wondering if you interested in these changes? 

We also implemented quite a few other changes, like making it easier to select a layout (shows English name, and can be searched), making the keyboard movable, and adding a help button.

I can make a pull request if you're interested. There are a lot of changes, so I wanted to at least ask before just making a pull request.

### PR DESCRIPTION
**Pull request by VitaJane**
Here are the changes for making the Korean Layout work.
  - Updated Unicode for Jamo characters used in the Korean layout to the current versions
  - Add algorithms and what not to handle Jamo input to get Hangul 
  - Added complete button to allow users to start a new Hangul character without it combining to the previous
 
Note: Hangul is the Korean writing system and Jamo are the individual characters that make up Hangul.

### CODE REVIEW

**Comment by GreyWyvern** [comment]
Okay, so first of all, this is great. I don't know how a Korean keyboard with these hangul-combining key sequences actually works, but your additions seem to work fine and without errors. It's apparent a great deal of work has gone into this, so great job!

Apart from some code styling and convention issues, the major concern I have is that a large part of the code (between the 'For Korean input' and 'End for Korean input') occurs outside the `(function() { ... })();` scope which means it's global for any other script on the page to conflict with. All of these variables and functions need to be moved inside the function scope and ideally have their names updated to reduce the risk of conflicts. I suggest a prefix of `VKI_KO_` for the Korean-related functions and variables.

As well, I realize that some of the additions cannot be moved since they add to the existing logic, however if the majority of added code could be grouped into one location, then users could easily remove it if they decide not to use the Korean keyboard and save on script size.

A few other things confuse me a bit, like how you have a function `kEvent()` that clears the `current` array and the Complete button content, but the function seems to be duplicated by the `clearCurrent()` function, and in several other places the `current` variable and Complete button are cleared manually, but neither of these functions is called.

Those things can wait until after the KO functions are moved into the keyboard function scope. Let me know if you'd like me to do that for you.

Thanks again for this PR!

**Comment by VitaJane** [comment]
I can go ahead and make those changes.

**Comment by VitaJane** [comment]
Alright, I made those changes you suggested. You were right, there were some things that were basically just doing the same thing so I fixed those.

File: keyboard.js [hunk 1]
```diff
@@ -1869,8 +2331,8 @@ var VKI_attach, VKI_close;
 
   function VKI_getStyle(obj, styleProp) {
     let y;
-    if (obj.currentStyle) {
-      y = obj.currentStyle[styleProp];
+    if (obj.VKI_KO_currentStyle) {
+      y = obj.VKI_KO_currentStyle[styleProp];
```
Comment (GreyWyvern) [inline keyboard.js]: One more change. This should remain just `currentStyle`.

**Comment by GreyWyvern** [comment]
Looks great! Everything else apart from code-style issues (I can tinker with those later) seems fine, however I did find a bug.

If I type in two or three Korean jamo that combine and do NOT press the Combine key, then I switch to another keyboard, then press the Backspace key, there is a javascript error.

Switching to another keyboard layout should probably complete the Korean character and reset the Complete button.

**Comment by VitaJane** [comment]
I went ahead and fixed the bug.

**Comment by GreyWyvern** [comment]
It seems fixed now, thanks! Just the `currentStyle` revert and I think it is ready to merge.

Any other edits you'd like to make here? Would you like to add your name and / or contact info in the file header? You can add it before the Github link, and make it something like: "Korean Jamo to Hangul input by ..."

This is a great addition to the script. I really appreciate it!

**Comment by VitaJane** [comment]
Yeah, I actually have a few more big changes I made. Made a tweak to the language dropdown so you could see the English names next to the native name (optionally) and made it searchable. Also added a help button and made the keyboard movable it's self. Don't know if you want these changes. And yeah, I'd love to add my name to the header.

**Comment by GreyWyvern** [comment]
> Yeah, I actually have a few more big changes I made. Made a tweak to the language dropdown so you could see the English names next to the native name (optionally) and made it searchable. Also added a help button and made the keyboard movable it's self. Don't know if you want these changes.

I meant related to this PR. I'd definitely like to see your other changes in separate PRs.

> And yeah, I'd love to add my name to the header.

Please do!

**Comment by VitaJane** [comment]
Sorry my bad, misunderstood. I went added to the header, that should be everything for the PR.

**Review by GreyWyvern — CHANGES_REQUESTED** [review]

**Comment by VitaJane** [comment]
Just fixed it

**Review by GreyWyvern — APPROVED** [review]

**Comment by GreyWyvern** [comment]
Thanks very much for all your hard work!

**Pull request merged**